### PR TITLE
[SYCL] Moved on-device tests from intel/llvm

### PR DESCRIPTION
**Pull request by vladimirlaz**
- The on-device tests for persistent cache moved;
 - Marked implicit_offset_debug_info.cpp XFAIl to avoid false alarms in
   CI.

### CODE REVIEW

**Comment by vladimirlaz** [comment]
SYCl/USM/pointer_query.cpp has been fixed in  https://github.com/intel/llvm/commit/691f8428ad87c27136e6aa2a01729b86d485c03e

File: SYCL/KernelAndProgram/spec_consts.hpp [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            << "\n";
+  std::cout << "global_val = " << global_val << "\n";
+  cl::sycl::program program1(q.get_context());
+  cl::sycl::program program2(q.get_context());
+  cl::sycl::program program3(q.get_context());
+  cl::sycl::program program4(q.get_context());
+
+  int goldi = (int)get_value();
+  // TODO make this floating point once supported by the compiler
+  float goldf = (float)get_value();
```
Comment (s-kanaev) [inline SYCL/KernelAndProgram/spec_consts.hpp]:
```suggestion
  const float goldf = (float)get_value();
```

File: SYCL/KernelAndProgram/spec_consts.hpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+  std::cout << "Running on " << q.get_device().get_info<info::device::name>()
+            << "\n";
+  std::cout << "global_val = " << global_val << "\n";
+  cl::sycl::program program1(q.get_context());
+  cl::sycl::program program2(q.get_context());
+  cl::sycl::program program3(q.get_context());
+  cl::sycl::program program4(q.get_context());
+
+  int goldi = (int)get_value();
```
Comment (s-kanaev) [inline SYCL/KernelAndProgram/spec_consts.hpp]:
```suggestion
  const int goldi = (int)get_value();
```

File: SYCL/KernelAndProgram/spec_consts.hpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  bool passed = true;
+  int vali = veci[0];
+
+  if (vali != goldi) {
+    std::cout << "*** ERROR: " << vali << " != " << goldi << "(gold)\n";
+    passed = false;
+  }
+  int valf = vecf[0];
+
+  if (valf != goldf) {
```
Comment (s-kanaev) [inline SYCL/KernelAndProgram/spec_consts.hpp]: Should the values here be checked for absolute difference?

**Review by s-kanaev — APPROVED** [review]

**Pull request merged**